src/publish.rs

```diff
@@ -254,18 +254,19 @@ pub fn publish_crate(mut body: Bytes, index: &mut Index) -> Result<()> {
     "crate name contains non-ASCII characters"
   );
 
-  let crate_meta_dir = index.root().join(crate_path(&crate_name));
+  let crate_meta_relative_dir = crate_path(&crate_name);
+  let crate_meta_dir = index.root().join(&crate_meta_relative_dir);
   create_dir_all(&crate_meta_dir)
     .with_context(|| format!("failed to create directory {}", crate_meta_dir.display()))?;
 
   let crate_length = parse_u32(&mut body)
     .context("failed to read crate length")?
-    .try_into()
-    .unwrap();
+    .try_into()?;
```

Comment:
Let's revert these changes, please. It's inconsistent with code higher up that still unwraps and on top of that any context is missing, so if we see such an error it will be less than useful.